### ISSUE TITLE
VAProfile is not supported (12) With MPEG2 playback

### ISSUE THREAD

**Issue by heitbaum** [intel/libva#477]
This is the error being logged using 2.9.0 and 2.10.0.pre2 playing an MPEG2 stream.
Any reason this is occurring and the VAAPI is not HW decoding, given that MPEG2 is listed as supported.

2020-12-02 23:31:20.528 T:1151     INFO <general>: VAAPI::Close - closing decoder context
2020-12-02 23:31:45.200 T:3650    ERROR <general>: VAAPI/vpp vaCreateConfig error: the requested VAProfile is not supported (12)
2020-12-02 23:31:45.206 T:3650    ERROR <general>: VAAPI output: Postproc preinit failed

LibreELEC:~ # vainfo
libva info: VA-API version 1.10.0
libva info: Trying to open /usr/lib/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_9
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.10 (libva 2.9.1)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 20.3.0 (a837220d2a)
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointEncSliceLP
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointEncSliceLP
      VAProfileJPEGBaseline           : VAEntrypointVLD
      VAProfileJPEGBaseline           : VAEntrypointEncPicture
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSliceLP
      VAProfileVP8Version0_3          : VAEntrypointVLD
      VAProfileHEVCMain               : VAEntrypointVLD

This is on the following CPU and the LATEST Mesa 
[    0.224877] smpboot: CPU0: Intel(R) Core(TM) i5-6260U CPU @ 1.80GHz (family: 0x6, model: 0x4e, stepping: 0x3)

2020-12-02 23:31:20.528 T:1151     INFO <general>: CRenderSystemGL::InitRenderSystem - Version: 4.6 (Core Profile) Mesa 20.2.3, Major: 4, Minor: 6
2020-12-02 23:31:20.528 T:1151     INFO <general>: GL_RENDERER = Mesa Intel(R) Iris(R) Graphics 540 (SKL GT3)
2020-12-02 23:31:20.528 T:1151     INFO <general>: GL_VERSION = 4.6 (Core Profile) Mesa 20.2.3

The media in question (ffprobe output) is:
Video: mpeg2video (Main), yuv420p(tv, top first), 720x576 [SAR 16:15 DAR 4:3], 25 fps, 25 tbr, 90k tbn, 50 tbc

**Comment by XinfengZhang** [comment]
seems a media-driver issue, create config failed. could you help to share more details, such as command line?

**Comment by heitbaum** [comment]
Hi @XinfengZhang - it has been happening for a long time in the Kodi Leah/Matrix time. I have done some investigation and brought all the Intel code up to current in The Matrix HEAD. These are the logs from cvh . http://ix.io/2HpU, these are the logs from me. http://ix.io/2HpA

My uneducated guess is that Kodi is calling ffmpeg with a profile=12... might be wrong (12 in the vadumpcaps is JPEG ???)

Here is the bug I raised on LibreELEC - https://github.com/LibreELEC/LibreELEC.tv/issues/4732

Here is the full background and investigation I have done so far - https://forum.libreelec.tv/thread/23090-intel-vaapi-and-hardware-mpeg2video-decoding/?postID=147266#post147266

**Comment by XinfengZhang** [comment]
yes, 12 is jpeg, please refer to https://github.com/intel/libva/blob/master/va/va.h#L484
